string leftX
string leftY
string rightX
string rightY